File: frontend/src/types/shift.ts
// backend/app/schemas/shift.py::ShiftOut ile birebir eşleşir.
export interface ShiftItem {
  employee_id: number;
  employee_name: string;
  shift_date: string;
  start_time: string | null;
  end_time: string | null;
  is_day_off: boolean;
}

// backend/app/schemas/shift.py::RosterEmployee ile birebir eşleşir.
export interface RosterEmployee {
  id: number;
  first_name: string;
  last_name: string;
  role: string;
}

// backend/app/schemas/shift.py::ShiftUpsert ile birebir eşleşir.
export interface ShiftUpsertPayload {
  shift_date: string;
  start_time?: string | null;
  end_time?: string | null;
  is_day_off: boolean;
}

File: backend/app/schemas/shift.py
from datetime import date, time

from pydantic import BaseModel, ConfigDict


class ShiftOut(BaseModel):
    employee_id: int
    employee_name: str
    shift_date: date
    start_time: time | None = None
    end_time: time | None = None
    is_day_off: bool


class ShiftUpsert(BaseModel):
    shift_date: date
    start_time: time | None = None
    end_time: time | None = None
    is_day_off: bool = False


class RosterEmployee(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    first_name: str
    last_name: str
    role: str
